src/main/java/com/getpcpanel/monitor/MonitorBrightnessService.java

```diff
@@ -0,0 +1,221 @@
+package com.getpcpanel.monitor;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+
+import com.getpcpanel.Json;
+import com.getpcpanel.spring.OsHelper;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import one.util.streamex.StreamEx;
+
+@Log4j2
+@Service
+@RequiredArgsConstructor
+public class MonitorBrightnessService {
+    private static final int COMMAND_TIMEOUT_MS = 4000;
+    private final OsHelper osHelper;
+    private final Json json;
+
+    public List<MonitorInfo> getMonitors() {
+        if (osHelper.notWindows()) {
+            return Collections.emptyList();
+        }
+
+        var nativeMonitors = MonitorNativeUtil.listMonitors();
+        if (!nativeMonitors.isEmpty()) {
+            return nativeMonitors;
+        }
+
+        var output = runPowerShell(buildListScript());
+        var jsonLine = extractJsonLine(output);
+        if (StringUtils.isBlank(jsonLine)) {
+            return Collections.emptyList();
+        }
+        try {
+            var trimmed = jsonLine.trim();
+            if ("null".equalsIgnoreCase(trimmed)) {
+                return Collections.emptyList();
+            }
+            if (trimmed.startsWith("{")) {
+                trimmed = "[" + trimmed + "]";
+            }
+            var monitors = json.read(trimmed, MonitorInfo[].class);
+            return StreamEx.of(Arrays.asList(monitors)).distinct(MonitorInfo::id).toList();
+        } catch (Exception e) {
+            log.warn("Failed to parse monitor list: {}", e.getMessage());
+            return Collections.emptyList();
+        }
+    }
+
+    public void setBrightness(String monitorId, int value) {
+        if (osHelper.notWindows()) {
+            return;
+        }
+        var brightness = Math.max(0, Math.min(100, value));
+        if (StringUtils.isBlank(monitorId)) {
+            for (var monitor : getMonitors()) {
+                setBrightnessSingle(monitor.id(), brightness);
+            }
+            return;
+        }
+        setBrightnessSingle(monitorId, brightness);
+    }
+
+    private void setBrightnessSingle(String monitorId, int value) {
+        if (StringUtils.isBlank(monitorId)) {
+            return;
+        }
+        if (monitorId.startsWith("dxva2:")) {
+            var index = parseDxva2Index(monitorId);
+            if (index >= 0) {
+                if (MonitorNativeUtil.setBrightnessByIndex(index, value)) {
+                    return;
+                }
+            }
+            return;
+        }
+        var id = escapeSingleQuotes(monitorId);
+        var script = """
```

Comment:
Don't put full powershell scripts within java source. Make it a resource.

Is there no native Java way of doing this? Otherwise, this would probably perform way better if built in the c-code instead of calling powershell scripts.